Shared/Samples/Display device location with NMEA data sources/DisplayDeviceLocationWithNMEADataSourcesView.Model.swift

```diff
@@ -0,0 +1,273 @@
+                guard let protocolString = accessory.protocolStrings.first(where: { supportedProtocolStrings.contains($0) }) else {
+                    // Skip the accessories with protocol not for NMEA data transfer.
+                    continue
+                }
+                // Only return the first connected and supported accessory.
+                return (accessory, protocolString)
+            }
+            return nil
+        }
+        
+        /// The Bluetooth accessory picker connected to a supported accessory.
+        func accessoryDidConnect(connectedAccessory: EAAccessory, protocolString: String) {
+            if let dataSource = NMEALocationDataSource(accessory: connectedAccessory, protocol: protocolString) {
+                nmeaLocationDataSource = dataSource
+                start()
+            }
+        }
+        
+        /// Reset the sample, stops the data source, cancels tasks, and resets button states and status strings.
+        func reset() {
+            // Reset buttons states.
+            isResetButtonDisabled = true
+            isSourceButtonDisabled = false
+            
+            // Reset the status text.
+            accuracyStatus = "Accuracy info will be shown here."
+            satelliteStatus = "Satellites info will be shown here."
+            
+            // Reset NMEA location data source.
+            nmeaLocationDataSource = nil
+            autoPanMode = .off
+
+            Task {
+                // Stop the location display, which in turn stop the data source.
+                await locationDisplay.dataSource.stop()
+            }
+            
+            // Cancel the autoPan, location, and satellite observation tasks.
+            cancelObservationTasks()
+            
+            // Pause the mock data generation.
+            mockNMEADataSource.stop()
+        }
+
+        /// Starts the location data source and awaits location and satellite updates.
+        /// - Parameter usingMockedData: Indicates that the location datasource should use mocked data.
+        func start(usingMockedData: Bool = false) {
+            // Set NMEA location data source for location display.
+            locationDisplay.dataSource = nmeaLocationDataSource
+            // Set buttons states.
+            isSourceButtonDisabled = true
+            isResetButtonDisabled = false
+            
+            // Start the data source and location display.
+            if usingMockedData {
+                mockNMEADataSource.start(with: nmeaLocationDataSource)
+            }
+            
+            // Set the autopan mode to `.recenter`
+            autoPanMode = .recenter
+
+            Task {
+                // Start the data source
+                try await locationDisplay.dataSource.start()
+            }
+            
+            // Kick off tasks to monitor autoPan, locations, and satellites.
+            observationTasks.append(
```

Comment:
After some initial suggestions are applied, I'll add Ryan or Phil R to take a look. From the doc, detached tasks are not the best choice if we have other options.

Reply:
I did try a TaskGroup but didn't have any success. The issue is that the observation of Locations and Satellites are never ending, so the task never has a chance to complete. If they are not detached tasks, I think they continue to hold onto self, and will never be released.